src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -99,6 +105,16 @@ private void initialize(NativeMemoryCacheManagerDto nativeMemoryCacheDTO) {
 
         if (nativeMemoryCacheDTO.isExpirationLimited()) {
             cacheBuilder.expireAfterAccess(nativeMemoryCacheDTO.getExpiryTimeInMin(), TimeUnit.MINUTES);
+            Runnable cleanUp = () -> {
+                try {
+                    cache.cleanUp();
+                } catch (Exception e) {
+                    logger.error("Error cleaning up cache", e);
```

Comment:
Why swallow exception here?

Reply:
Any exceptions from Guava cache operations would otherwise halt our scheduled executor. If an exception occurs here, it would be from Guava's cache manager internals rather than our logic, so per [Dooyong's suggestion](https://github.com/opensearch-project/k-NN/pull/2308#discussion_r1893549051) above we can log it and continue scheduling cleanup tasks.

This way we can ensure the cache maintenance keeps running even if individual cleanup attempts fail, and we can monitor Guava errors in the logs.